NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -339,6 +359,47 @@ class NeuroID
             }
         }
 
+        /**
+         * ported from the iOS implementation
+         */
+        override fun startAppFlow(linkedSiteId: String, userID: String?): SessionStartResult {
+            // reset linked site id (in case of failure)
+            linkedSiteID = ""
+            if (!verifyClientKeyExists() || !validateSiteId(linkedSiteId)) {
+                return SessionStartResult(false, "")
+            }
+
+            // immediately flush events before anything else
+            CoroutineScope(Dispatchers.IO).launch {
+                nidJobServiceManager.sendEvents(true)
+                saveIntegrationHealthEvents()
+            }
+            // If not started then start
+            var startStatus: SessionStartResult = if (!isSDKStarted) {
+                // if userID passed then startSession else start
+                if (userID != null) {
+                    startSession(userID)
```

Comment:
will this just return the result from startSession I am assuming? (not the most familiar with kotlin syntax)

Reply:
yep, just sets startStatus with the SessionStartResult from startSession call, start() or whatever else is returns a startSession and the failure case.